Gem/Code/Source/ShaderReloadTestComponent.cpp

```diff
@@ -122,22 +126,51 @@ namespace AtomSampleViewer
         }
     }
 
-    bool ShaderReloadSoakTestComponent::ReadInConfig(const AZ::ComponentConfig*)
+    bool ShaderReloadTestComponent::ReadInConfig(const AZ::ComponentConfig*)
     {
         return true;
     }
 
-    void ShaderReloadSoakTestComponent::Activate()
+    void ShaderReloadTestComponent::Activate()
     {
         AZ::TickBus::Handler::BusConnect();
 
         InitTestDataFolders();
+
+        DeleteTestFile("Fullscreen.shader");
+        DeleteTestFile("Fullscreen.azsl");
```

Comment:
In MaterialHotReloadTest I had to wait for the assets to fully disappear from the asset manager before continuing. Is that not needed here? See MaterialHotReloadTestComponent's InitStatus::ClearingTestAssets.

Reply:
The way AssetStatusTracker works, it may be necessary to guarantee that the assets are fully deleted, but I can do that check during Activate(). The idea is that you can stop the debugger and never call Deactivate(), so next time around if this case is not considered inside Activate() it will be a guaranteed failure.